sql/sql_show.cc

```diff
@@ -2031,11 +2038,6 @@ int show_create_table(THD *thd, TABLE_LIST *table_list, String *packet,
       packet->append(STRING_WITH_LEN(")"));
     }
 
-    if (field->comment.length)
-    {
-      packet->append(STRING_WITH_LEN(" COMMENT "));
-      append_unescaped(packet, field->comment.str, field->comment.length);
-    }
     append_create_options(thd, packet, field->option_list, check_options,
                           hton->field_options);
```

Comment:
`append_create_options` should also be before the constraint. And please add a test case for it.

Reply:
@vuvova I have updated.

Reply:
I don't understand your new test case, sorry.
Are you testing whether create options are shows in `SHOW CREATE TABLE`?
There was no need to. First, this is already tested elsewhere. And second — you did not change it, so you are not testing your changes.

You've changed the order in which create options are printed as compare to `CHECK` constraint. This is what you should test. For example,
```
create table t1 (f int foo=bar check(f>0));
show create table t1;
```